honeybee_vtk/tovtk.py

```diff
@@ -0,0 +1,311 @@
+
+    return grouped_points
+
+
+def create_polygon(points):
+    """Create a vtk Polygon from a list of points.
+
+    The point will have to be in an order. This function cannot be used to create a
+    polygon with holes.
+
+    Args:
+        points: A list of points. Here, each point is a list of X, Y, and Z cordinate.
+
+    Returns:
+        A tuple with two elements.
+
+        - vtk_points: A list of vtk point objects
+
+        - vtk_polygon: A vtk polygon object created from the points provided.
+    """
+    vtk_points = vtk.vtkPoints()
+    vtk_polygon = vtk.vtkPolygon()
+    vtk_polygon.GetPointIds().SetNumberOfIds(len(points))
+
+    for point in points:
+        vtk_points.InsertNextPoint(tuple(point))
+
+    for i in range(len(points)):
+        vtk_polygon.GetPointIds().SetId(i, i)
+
+    return vtk_points, vtk_polygon
+
+
+def create_polygons(points_lst):
+    """Create a vtk object with multiple vtk polygons.
+
+    Args:
+        points_lst: A list of lists of points. Here, each point is a list of X, Y, 
+            and Z cordinate.
+
+    Returns:
+        A vtk object that with multiple polygons.
+    """
+    vtk_polydata_lst = []
+
+    for i in range(len(points_lst)):
+
+        vtk_points, vtk_polygon = create_polygon(points_lst[i])
+        vtk_polygons = vtk.vtkCellArray()
+        vtk_polygons.InsertNextCell(vtk_polygon)
+
+        vtk_polydata = vtk.vtkPolyData()
+        vtk_polydata.SetPoints(vtk_points)
+        vtk_polydata.SetPolys(vtk_polygons)
+        vtk_polydata.Modified()
+
+        vtk_polydata_lst.append(vtk_polydata)
+
+    vtk_polydata_extended = vtk.vtkAppendPolyData()
+    
+    for i in range(len(vtk_polydata_lst)):
+        vtk_polydata_extended.AddInputData(vtk_polydata_lst[i])
+
+    vtk_polydata_extended.Update()
+
+    return vtk_polydata_extended
+
+
+def write_polydata(polydata, file_name):
+    """Write a VTK file to the disk.
+
+    Args:
+        polydata: An appended VTK polydata object.
+        file_name: A text string for the the file name to be written.
+    """
+    vtk_polydata_extended = create_polygons(polydata)
+    writer = vtk.vtkPolyDataWriter()
+    writer.SetFileName(file_name + '.vtk')
+    writer.SetInputConnection(vtk_polydata_extended.GetOutputPort())
+    writer.Write()
+
+
+def write_color_grouped_points(points, vectors, file_name='grid points'):
+    """Write color-grouped VTK points to disk.
+
+    Args:
+        points : A list of points. Here, each points is a list of X, Y, and Z
+            cordinates of the point.
+        vectors: A list of grid vectors where each vector is a list of X, Y, and Z
+            component of a vector.
+        file_name: A text string to be used as a file name. Defaults to "grid points."
+    """
+    point_polydata = point_vectors(points, vectors)
+    writer = vtk.vtkPolyDataWriter()
+    writer.SetFileName(file_name + '.vtk')
+    writer.SetInputData(point_polydata)
+    writer.Write()
```

Comment:
Maybe it will be nice to return the file path from these write functions.

Reply:
Okay. Can you please share what can we do with those file paths?

Reply:
It will be easier to use. Let's say we are developing a command line tool that uses this function and we want to access the file after its being written. It will be handy to get the filepath from this function instead of trying to create it. It will also ensure changing the path/name will not break any downstream code.

Reply:
Done